File: main/java/org/goafabric/core/data/repository/entity/AddressEo.java
package org.goafabric.core.data.repository.entity;

import jakarta.persistence.*;
import org.goafabric.core.data.repository.extensions.AuditListener;


@Entity
@Table(name="address")
@EntityListeners(AuditListener.class)
public class AddressEo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    public String id;

    public String use;

    public String street;
    public String city;

    public String postalCode;
    public String state;
    public String country;

    @Version //optimistic locking
    public Long version;
}

File: src/main/kotlin/org/goafabric/core/organization/persistence/entity/AddressEo.kt
package org.goafabric.core.organization.persistence.entity

import jakarta.persistence.*
import org.goafabric.core.persistence.extensions.AuditTrailListener
import org.goafabric.core.persistence.extensions.KafkaPublisher

@Entity
@Table(name = "address")
@EntityListeners(AuditTrailListener::class, KafkaPublisher::class)
class AddressEo(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null,

    var use: String? = null,
    var street: String? = null,
    var city: String? = null,
    var postalCode: String? = null,
    var state: String? = null,
    var country: String? = null,

    @Version
    var version: Long? = null
)
